### PR TITLE
fix(stack): extend as prop from BoxProps on StackProps

### PR DESCRIPTION
**Pull request by richbachman**
- [x] Extend `as` prop from `BoxProps`
- [x] Add tests

### CODE REVIEW

File: packages/paste-core/layout/stack/__tests__/stack.test.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import {render as testRender} from '@testing-library/react';
+import {render as testRender, screen} from '@testing-library/react';
```
Comment (SiTaggart) [inline packages/paste-core/layout/stack/__tests__/stack.test.tsx:2]: Are you using `screen` as all?

Comment (SiTaggart) [reply packages/paste-core/layout/stack/__tests__/stack.test.tsx:2]: I don't think you need to re assign render as testRender here either

Comment (richbachman) [reply packages/paste-core/layout/stack/__tests__/stack.test.tsx]: Tried to use a `screen` method to test as a `header`, but it wasn't working as I thought. Just forgot to remove it.

File: packages/paste-core/layout/stack/__tests__/stack.test.tsx [hunk 2]
```diff
@@ -99,4 +110,9 @@ describe('Stack', () => {
     const {asFragment} = testRender(<MockResponsiveStack />);
     expect(asFragment()).toMatchSnapshot();
   });
+
+  it('should render as a header', () => {
+    const {asFragment} = testRender(<MockHeaderStack />);
+    expect(asFragment()).toMatchSnapshot();
```
Comment (SiTaggart) [inline packages/paste-core/layout/stack/__tests__/stack.test.tsx:116]: I think for these I would like you to assert the element outside of a snapshot. And also provide some tests for arbitary html attributes

Comment (richbachman) [reply packages/paste-core/layout/stack/__tests__/stack.test.tsx]: I changed that test and added a couple more for valid attributes and one for the invalid className attribute. Let me know if I should add more.

File: packages/paste-core/layout/stack/__tests__/stack.test.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import {render} from '@testing-library/react';
+import {render, screen} from '@testing-library/react';
```
Comment (SiTaggart) [inline packages/paste-core/layout/stack/__tests__/stack.test.tsx:2]: Still importing screen

File: packages/paste-core/layout/stack/__tests__/stack.test.tsx [hunk 4]
```diff
@@ -112,7 +120,22 @@ describe('Stack', () => {
   });
 
   it('should render as a header', () => {
-    const {asFragment} = render(<MockHeaderStack />);
-    expect(asFragment()).toMatchSnapshot();
+    const {getByTestId} = render(<MockHeaderStack />);
+    expect(getByTestId('header')).not.toBeNull();
```
Comment (SiTaggart) [inline packages/paste-core/layout/stack/__tests__/stack.test.tsx:124]: You need to test the element tag name or something here. This only tests that it exists by test id. If I change the as prop, the test will still pass

File: packages/paste-core/layout/stack/__tests__/stack.test.tsx (hunk 3 above)
Comment (richbachman) [reply packages/paste-core/layout/stack/__tests__/stack.test.tsx]: Finally remove.

File: packages/paste-core/layout/stack/__tests__/stack.test.tsx (hunk 4 above)
Comment (richbachman) [reply packages/paste-core/layout/stack/__tests__/stack.test.tsx]: Changed that to test for the actual element.

**Pull request merged**